forest/daemon/src/daemon.rs

```diff
@@ -427,15 +427,22 @@ async fn prompt_snapshot_or_die(config: &Config) -> bool {
                 )
                 .default(false)
                 .interact()
-                .unwrap_or_default()
     } else {
-        config.client.download_snapshot
+        Ok(config.client.download_snapshot)
     };
 
-    if should_download {
-        true
-    } else {
-        cli_error_and_die("Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file] or --download-snapshot to download one automatically", 1);
+    match should_download {
+        Ok(result) => {
+            if result {
+                true
+            } else {
+                cli_error_and_die("Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file] or --download-snapshot to download one automatically", 1);
+            }
+        }
+        Err(_) => {
+            std::mem::drop(should_download);
+            cli_error_and_die("Error getting input.", 1);
+        }
```

Comment:
Are these changes related to the cursor fix? If so, could you explain it in a comment. If not then you can delete this code.

Reply:
Yes, this was originally simply part of my troubleshooting, but I just tested it without the match statement, and it is required to properly handle the prompt resuts including the case of an interrupt. I'll add a comment.